miden-lib/src/transaction/inputs.rs

```diff
@@ -290,3 +303,24 @@ fn add_input_notes_to_advice_inputs(
     // NOTE: keep map in sync with the `prologue::process_input_notes_data` kernel procedure
     inputs.extend_map([(tx_inputs.input_notes().commitment(), note_data)]);
 }
+
+// KERNEL HASHES INJECTOR
+// ------------------------------------------------------------------------------------------------
+
+/// Inserts kernel hashes and hashes of their procedures into the provided advice inputs.
+///
+/// Inserts the following entries into the advice map:
+/// - The accumulative hash of all kernels |-> array of each kernel hash.
+/// - The hash of the selected kernel |-> array of the kernel's procedure hashes.
+pub fn add_kernel_hashes_to_advice_inputs(inputs: &mut AdviceInputs, kernel_offset: u8) {
+    let kernel_hashes = [TransactionKernel::kernel_hash(kernel_offset).as_elements()].concat();
```

Comment:
The idea was to prepare code to be able to manage several kernel versions, so in case of two versions this variable will be calculated like so:
```rust
let kernel_hashes = [TransactionKernel::kernel_hash(0).as_elements(), TransactionKernel::kernel_hash(1).as_elements()].concat();
```
so `concat()` here is needed to create a `Felt` array instead of array of arrays of `Felt`s. But now, looking at it with a fresh look, I think that for now we can simplify this, since almost certainly this code will be reworked when we add a kernel versioning support.